### ISSUE TITLE
FBI v2.5.3 Will Not Download Newest PKSM File.

### ISSUE THREAD

**Issue by WarriorCatBluestar** [Steveice10/FBI#533]
Hello, I am rather new to this whole FBI, CIA, and PKSM file thing. One of my close friends allowed me to access PKSM on my 3DS a while ago, probably sometime in 2018. Everything ran smoothly, nothing bad to report. The issue now is, it’s 2020, and there have been some major updates to PK. My current version is v5.1.4, which is severely outdated compared to the new v8.0.3. In order to update it, I was told to go to the FBI app on my 3DS, and go to Remote Install. I went there, selected the QR code option, and tried to scan the QR for the PK update. It didn’t work, and gave me this error on the bottom screen:

Failed to install from URL.
https://github.com/FlagBrew/PKSM/releases/download/8.0.3/PKSM.cia
Result code: 0xD8A0A03C
Level: Permanent (27)
Summary: Invalid state (5)
Module: HTTP (40)
Desc: Failed to verify TLS certificate (60)

Nothing except the PKSM url is at the top screen. I’ve heard there are some issues with files called .cia files, so I’m expecting the reason to be because of that? Again, I’m not too sure as I quite literally just started looking at the files today. I thought naturally that I should update FBI since it’s at v2.5.3 and I’m not sure if it’s outdated, but when I tried to update it it gave me this error:

Failed to check for update.
Result code: 0xD963F9FE
Level: Permanent (27)
Summary: Internal (11)
Module: Application (254)
Desc: HTTP 502: Bad Gateway (510)

My internet is fine and everything is how it should be. I’m not sure if it matters, but if it helps you, the Luma3DS Updater won’t update either, and gives me this error:

FATAL: Could not get Location header for 3xx reply
Ret code: ded92

Sorry for having all of these issues, I’m just a noob at this I guess 😂. I wasn’t exactly sure of where to post this either, since I’m new to Github and this kinda includes more programs than one, so sorry if someone needs to move this. I think FBI might be the main issue here though. If any of these issues require one of those SD micro card reader things, then do let me know so I can get one. I’ve also heard that deleting FBI then reinstalling it helps, but I’m kinda scared to do that since I don’t actually see a CIA app (because I think you have to reinstall it from there, right?) on my 3DS and I might not get FBI back. Thank you for reading, and I appreciate any help! Oh, and my handheld is a New 3DS XL and it’s system is at 11.13.0-45U if that helps or has anything to do with it.

**Comment by Steveice10** [comment]
GitHub dropped support for the version of TLS supported by the 3DS, and FBI's updater has changed URLs. If you download and install the latest FBI from the releases page, both the updater and your PKSM download should work again.

**Comment by WarriorCatBluestar** [comment]
> GitHub dropped support for the version of TLS supported by the 3DS, and FBI's updater has changed URLs. If you manually download and install the latest FBI from the releases page, both the updater and your PKSM download should work again.

I do apologize, however I've gotta ask how exactly to do that? I'm using a Chromebook, but I can use a Mac if that's more applicable. I'm, I guess you could say, at the part where I've downloaded the files onto my Chromebook, and I've recently found an adapter for my 3DS chip, so I'm able to see what's on my SD card on here. I can see the files, but now I'm not sure where to put them. If you could 'dumb it down' for me that'd be great XD!

**Comment by Steveice10** [comment]
You'll want to put the downloaded FBI.cia on your SD card, open up FBI on your 3DS, find it in the SD card menu, and install it.

**Comment by WarriorCatBluestar** [comment]
> You'll want to put the downloaded FBI.cia on your SD card, open up FBI on your 3DS, find the CIA file in the SD card menu, and install it.

Do I 'Install And Delete CIA' or just 'Install'? I clicked 'Install" and tried the QR code for PKSM, which didn't work, but then I downloaded the PKSM.cia and went to the home menu, both the FBI and PKSM icons disappeared for a second, then came back. I tried PK, and it loaded, but then it blacked out my bottom screen and on the top was the PK icon bouncing around the screen like how the old DVD icon did. I had my Ultra Sun game in it, so it should've worked.
